plugin/pkg/shoot/dnshostedzone/admission.go

```diff
@@ -174,6 +175,17 @@ func verifyHostedZoneID(shoot *garden.Shoot, secretBindingLister gardenlisters.S
 		if !serviceAccountJSONFound {
 			return fmt.Errorf("specifying the `.spec.dns.hostedZoneID` field is only possible if the cloud provider secret or the secret referenced in .spec.dns.secretName contains credentials for Google CloudDNS (%s)", gcpbotanist.ServiceAccountJSON)
 		}
+	case garden.DNSOpenstackDesignate:
```

Comment:
Could you please add a test for the OpenStack case (similar to [that](https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/dnshostedzone/admission_test.go#L210) and [that](https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/dnshostedzone/admission_test.go#L175))?